examples/algorithms/using_existing/linspace.cpp

```diff
@@ -0,0 +1,89 @@
+//==================================================================================================
+/**
+   EVE - Expressive Vector Engine
+   Copyright : EVE Contributors & Maintainers
+   SPDX-License-Identifier: MIT
+**/
+//==================================================================================================
+#include <eve/module/core.hpp>
+#include <eve/algo.hpp>
+#include "eve/algo/transform.hpp"
+#include <vector>
+
+//
+// In this example, we implements the linspace function. linspace is a staple of numeric computation
+// that, for a given pair of real [a,b], linspace(r,a,b) fills the range r with values so that
+//
+//  - the first element of r is equal to a;
+//  - the last element of r is equal to b;
+//  - each other elements are separated by (b-a)/(s.size()-1)
+//
+//  Precision is usually the cornerstone of proper linspace implementation. We use eve::lerp
+//  to ensure proper accuracy.
+//
+//  A small set of corner cases exists, and we decided to follow matlab semantic for those.
+//
+template < eve::algo::relaxed_range R, eve::scalar_value T,  eve::scalar_value U>
+R linspace(R& r, T a, U b )
+requires( eve::floating_scalar_value<decltype(eve::read(r.begin()))>)
+{
+  EVE_ASSERT(eve::is_finite(a), "a is not finite");
+  EVE_ASSERT(eve::is_finite(b), "b is not finite");
+
+  using elt_t = decltype(eve::read(r.begin()));
+  std::ptrdiff_t size = r.end()-r.begin();
+
+  // Empty range needs nothing to be done
+  if (size == 0u) return r;
+  // Size 1 range is just [b] as per Matlab
+  else if (size == 1u)  { *r.begin() = elt_t(b);        return r; }
+  // Similar begin and end values simplify to fill
+  else if (a == b)      { eve::algo::fill(r, elt_t(a)); return r; }
+  else
+  {
+    // Computes 1/(size-1)
+    elt_t step  = eve::rec((elt_t(size - 1)));
+
+    // Make a view over [0 .... size-1]
+    auto io     = eve::views::iota(elt_t{0}, size);
+
+    // Compute lerp(a,b, iota[i] * step) for the whole range
+    eve::algo::transform_to ( eve::views::zip(io, r)
+                            , [a, b, step](auto i)
+                              {
+                                return eve::lerp(elt_t(a), elt_t(b), i*step);
+                              }
+                            );
```

Comment:
If we just divide b-a / step and then add it, it will give a much worse precision?

Reply:
for float it's terrible. for all types, you may end up with non exactly representable intervals that lead to not being even close to b in the end or even not incrementing.